### ISSUE TITLE
[Ajax / Server Error for action upgradeDb] textStatus: "error " errorThrown:"Internal Server Error " jqXHR: " "

### ISSUE THREAD

**Issue by Klemart3D** [PrestaShop/autoupgrade#120]
**Describe the bug**
During DB update, after core files successful updated:
`[Ajax / Server Error for action upgradeDb] textStatus: "error " errorThrown:"Internal Server Error " jqXHR: " "`

**Upgrade environment (please complete the following information):**
 - Version you started from: 1.7.4.1
 - Version you upgrade to: 1.7.4.2
 - Way of upgrade: Web
 - Step concerned: upgrade DB

**Displayed log**
```
[error] 5183#0: *10971 FastCGI sent in stderr: "PHP message: PHP Warning:  count(): Parameter must be an array or an object that implements Countable in /var/www/prestashop/html/modules/autoupgrade/AdminSelfUpgrade.php on line 3665" while reading response header from upstream

[error] 5180#0: *11061 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Symfony\Component\Debug\Exception\FatalThrowableError: Call to a member function can() on null in /var/www/prestashop/html/src/Adapter/Module/AdminModuleDataProvider.php:172

Stack trace:
#0 /var/www/prestashop/html/src/Core/Addon/Module/ModuleManager.php(391): PrestaShop\PrestaShop\Adapter\Module\AdminModuleDataProvider->isAllowedAccess('enable', 'ps_mainmenu')
#1 /var/www/prestashop/html/src/Core/Addon/Theme/ThemeManager.php(302): PrestaShop\PrestaShop\Core\Addon\Module\ModuleManager->enable('ps_mainmenu')
#2 /var/www/prestashop/html/src/Core/Addon/Theme/ThemeManager.php(176): PrestaShop\PrestaShop\Core\Addon\Theme\ThemeManager->doEnableModules(Array)
#3 /var/www/prestashop/html/modules/autoupgrade/AdminSelfUpgrade.php(2688): PrestaShop\PrestaShop\Core\Addon\Theme\ThemeManager->enable('classic')
#4 /var/www/prestashop/html/modules/autoupgrade/AdminSelfUpgrade.php(1996): AdminSelfUpgrade->doUpgrade()
#5 /var/www/prestashop/html/admin" while reading response header from upstream, client: 192.168.1.56, server: prestashop.dev, request: "POST /admin/autoupgrade/ajax-upgradetab.php HTTP/1.1", upstream: "fastcgi://unix:/var/php7-fpm/prestashop.sock:", host: "prestashop.dev", referrer: "http://prestashop.dev/admin/index.php?tab=AdminSelfUpgrade&token=026cc0b04393fd28021f93ef066b86b6"
```

Line 172 of AdminModuleDataProvider.php is:
```php
return ($this->employee->can('edit', 'AdminModulessf') && $this->moduleProvider->can('configure', $name));
```
Seems to be `$this->moduleProvider` is null?

**Expected behavior**
DB update must be successful

**Server (please complete the following information):**
 - Web Server: Nginx 1.6.2 
 - PHP Version: 7.2
 - Module Version 3.0.5

**Comment by Klemart3D** [comment]
Everything works fine after upgrade the module to v4.0.0.

**Comment by PierreRambaud** [comment]
Hey, thanks for this awesome feedback.

I close the pull request if everything is ok with the latest version.
Don't hesitate to come back to us if you have a problem :)

Kind Regards

**Comment by savgrace** [comment]
I recently upgraded from 1.7.2 to 1.7.5  and ran into this error. 
Checked my server logs through cPanel and noticed a "SoftException in Application.cpp is writeable by group"
Basically autoupgrade was halting during the upgrade process and it was a result of bad permissions in the root autoupgrade module directory.
Solution was to change the permissions on auto-upgradetab.php to 644.  
Hope someone finds that helpful.
